### ISSUE TITLE
400 Error during set_bearer() and get_all_tweets related to file/directory?

### ISSUE THREAD

**Issue by graceleffler** [cjbarrie/academictwitteR#183]
**Describe the bug**
set_bearer() is not allowing me to set the bearer token due to file/directory errors? Alternative attempts to set token within get_all_tweets request acts like there is missing text in place of bearer token.

**Expected behavior**
I expected 1. set_bearer() to quickly acknowledge the bearer token (rather than say there was no file/directory to save it in) and 2. embedded bearer token references to allow me to query data.

1. My first try was to use set_bearer()
Steps:
A. I ran the function "set_bearer()"
   i. .Renviron appeared in different tab. Already, the following was loaded into the console:
"TWITTER_PAT=/Users/**MYCOMPUTERUSERNAMEHERE**/.rtweet_token.rds
GOOGLE_MAPS_KEY=bul"

B. I saved my token: "TWITTER_BEARER <- MYTOKEN"
  i. Immediate error message appears (I clipped it; it repeats identical lines several times)

fit
**MYUSERNAME**-MBP:Class **MYUSERNAME**$ fit
bash: fit: command not found
**MYUSERNAME**-MBP:Class **MYUSERNAME**$ **MYUSERNAME**-MacBook-Pro:Class **MYUSERNAME**$ for(i in 1:12){
bash: syntax error near unexpected token `('
**MYUSERNAME**-MacBook-Pro:Class **MYUSERNAME**$ resgrm.pers<-factor.scores.grm(fit1, resp.patterns = NULL,
it1, resp.patterns = NULL, **MYUSERNAME**$ resgrm.pers<-factor.scores.grm(f
bash: syntax error near unexpected token `('
**MYUSERNAME**-MacBook-Pro:Class **MYUSERNAME**$ **MYUSERNAME**-MBP:Class **MYUSERNAME**$ TWITTER_BEARER <-**MYTOKEN**
bash: -: No such file or directory

C. I restarted R to follow directions despite the error.

D. I ran get_bearer(). It gives an error message about not having set the bearer token.

2. My second attempt tried to use an embedded bearer token within tweet request.
  A. Ran following code
  get_all_tweets(
    query = "covid-19",
    start_tweets = "2020-04-01T00:00:00Z",
    end_tweets = "2021-06-31T00:00:00Z",
    file = "vaccinations",
    data_path = "data/",
    n = 1,
    place="London",bearer_token = "MYTOKEN",
    point_radius =c(51.5073509,-0.1277583,24),lang="en")
 B. Saved BEARER_TOKEN <- MYTOKEN
 C. Ran embedded variable name in place of bearer_token
  get_all_tweets(
    query = "covid-19",
    start_tweets = "2020-04-01T00:00:00Z",
    end_tweets = "2021-06-31T00:00:00Z",
    file = "vaccinations",
    data_path = "data/",
    n = 1,
    place="London",bearer_token = BEARER_TOKEN,
    point_radius =c(51.5073509,-0.1277583,24),lang="en")

Resulting error:
"Error in make_query(url = endpoint_url, params = params, bearer_token = bearer_token,  : 
  something went wrong. Status code: 400
In addition: Warning message:
Directory already exists. Existing JSON files may be parsed and returned, choose a new path if this is not intended."

**Session Info:**
sessionInfo()
R version 3.6.3 (2020-02-29)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Sierra 10.12.5

Matrix products: default
BLAS:   /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib

Random number generation:
 RNG:     Mersenne-Twister 
 Normal:  Inversion 
 Sample:  Rounding 
 
locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] academictwitteR_0.2.0

loaded via a namespace (and not attached):
 [1] rstudioapi_0.13  magrittr_2.0.1   usethis_1.6.3    bit_4.0.4       
 [5] rjson_0.2.20     R6_2.5.0         rlang_0.4.9      fansi_0.4.1     
 [9] httr_1.4.2       tools_3.6.3      cli_2.2.0        DBI_1.1.0       
[13] ellipsis_0.3.1   yaml_2.2.1       bit64_4.0.5      assertthat_0.2.1
[17] tibble_3.0.4     lifecycle_0.2.0  crayon_1.3.4     twitteR_1.1.9   
[21] vctrs_0.3.5      fs_1.5.0         curl_4.3         glue_1.4.2      
[25] compiler_3.6.3   pillar_1.4.7     pkgconfig_2.0.3

**Comment by chainsawriot** [comment]
1. First, not a bug.
2. Please install the Github version

```r
devtools::install_github("cjbarrie/academictwitteR", build_vignettes = TRUE) 
```

3. Follow exactly what the `set_bearer()` says:

```
Add line: TWITTER_BEARER=YOURTOKENHERE to .Renviron on new line, replacing YOURTOKENHERE with  actual bearer token
```

Therefore, NOT `TWITTER_BEARER <- MYTOKEN`. Save your `.Renviron` and restart R.

Test and see if you can see your bearer token with `get_bearer()`

If it works, you don't need to specify the bearer_token anymore, i.e. 
```r
get_all_tweets(query = "covid-19", start_tweets = "2020-04-01T00:00:00Z", end_tweets = "2021-06-31T00:00:00Z")
```

**Comment by graceleffler** [comment]
It continues after I install the package.
MYTOKEN is shorthand for my bearer token.

**Comment by chainsawriot** [comment]
Have you done what the `set_bearer()` told you? i.e. adding this (or in your case, editing) to your .Renviron 

```
TWITTER_BEARER=AAAAAAAAAAAAAAAAAAAASJAHSnasbanmsbansbansbamnjkhajfhajdkfa
```

Of course, replace the gibberish with your own bearer token.

**Comment by graceleffler** [comment]
Yes, I did follow the set_bearer() directions and saved my bearer token as TWITTER_BEARER (step 1B above).

**Comment by chainsawriot** [comment]
I believe you have done that. Please note the equal sign; not arrow.
Then what did `get_bearer()` tell you? Did it return your token?

**Comment by graceleffler** [comment]
It has the same error for both the arrow and equal sign. The get_bearer() results in an error message asking me to set the bearer token: "Error: Please set envvar TWITTER_BEARER or supply your bearer token in every call. See ?get_bearer for more information."

**Comment by cjbarrie** [comment]
Did you Restart R after setting it?

**Comment by graceleffler** [comment]
Yes, I restarted R as in 1C.

**Comment by justinchuntingho** [comment]
Could you share what your .Renviron looks like (masking your actual bearer token) after pasting the token?

**Comment by graceleffler** [comment]
<img width="769" alt="Screen Shot 2021-07-05 at 6 41 05 PM" src="https://user-images.githubusercontent.com/70953983/124522365-03f8dd80-ddc1-11eb-8be9-f1590849eea2.png">

This is the very end of the long error message. The beginning reads as (and it repeats throughout):
"fit
MYUSERNAME-MBP:Class MYUSERNAME$ fit
bash: fit: command not found
MYUSERNAME-MBP:Class MYUSERNAME$ MYUSERNAME-MacBook-Pro:Class MYUSERNAME$ for(i in 1:12){
bash: syntax error near unexpected token (' **MYUSERNAME**-MacBook-Pro:Class **MYUSERNAME**$ resgrm.pers<-factor.scores.grm(fit1, resp.patterns = NULL, it1, resp.patterns = NULL, **MYUSERNAME**$ resgrm.pers<-factor.scores.grm(f bash: syntax error near unexpected token ('
MYUSERNAME-MacBook-Pro:Class MYUSERNAME$ MYUSERNAME-MBP:Class MYUSERNAME$ TWITTER_BEARER <-MYTOKEN"

**Comment by chainsawriot** [comment]
No spaces around the =.

**Comment by graceleffler** [comment]
I tried no spaces around the "=" both with quotation marks and without quotation marks; it doesn't seem to be helping or changing the error message.

**Comment by chainsawriot** [comment]
I actually suggest reading this first: Please [read the WTF guide on environmental variables](https://rstats.wtf/r-startup.html).

If you don't want to read and I try to make the long story extremely short: if you have this line in your `.Renviron`

```
ABC=123
```

and you run `Sys.getenv("ABC")` in your R console, you should get 123. In the same token, if you run `Sys.getenv("GOOGLES_MAP_KEY")` (with your current `.Renviron`), you should get "BUL". `get_bearer()` is actually just `Sys.getenv("TWITTER_BEARER")`

Could you please do this:

1. Run this in your Console TAB: `usethis::edit_r_environ()` (NOT THE TERMINAL TAB)
2. It will open up a new file, that is your `.Renviron`
3. INSERT THE LINE WITH NOT SPACE, USING EQUAL SIGH
4. Save your `.Renviron` (File -> Save)
5. Restart your R (Session -> Restart R)
6. Run this in your Console TAB: `academictwitteR::get_bearer()` 

If you still can't make it work, please supply the bearer token in every call.

**Comment by adriantanjm** [comment]
Hi! I faced the same problem as well after running the commands as suggested. I got the following error message: 

```
Error in make_query(url = endpoint_url, params = params, bearer_token = bearer_token,  : 
  something went wrong. Status code: 400
In addition: Warning messages:
1: Recommended to specify a data path in order to mitigate data loss when ingesting large amounts of data. 
2: Tweets will not be stored as JSONs or as a .rds file and will only be available in local memory if assigned to an object. 

```
Any suggestion to help with this?

**Comment by cjbarrie** [comment]
It's not clear this is the same issue. What code did you use to give this error? We cannot diagnose without this further information. Please follow the template for posting an Issue in future

**Comment by RosieBaillie** [comment]
I appear to be having the same issue. I have set TWITTER_BEARER in .Renviron; Sys.getenv("TWITTER_BEARER") and get_bearer() returns my bearer token correctly.

When I run this: 
`tweets <-
  get_all_tweets(
    "MY SEARCH TERM",
    "2019-01-01T00:00:00Z",
    "2021-05-07T00:00:00Z",
    get_bearer(),
    data_path = "MY DATA PATH",
    n = 100
  )`

I get the following message:

`Error in make_query(url = endpoint_url, params = params, bearer_token = bearer_token,  : 
  something went wrong. Status code: 400`

This was working for me when I first tried it out in May.
